### PR TITLE
[ADF-3282] Start task component refactored

### PR DESCRIPTION
**Pull request by davidcanonieto**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-3282

**What is the new behaviour?**
Added max name length, default task name, improved logic overall

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-3282

### CODE REVIEW

File: lib/process-services/task-list/components/start-task.component.ts [hunk 1]
```diff
@@ -45,6 +45,12 @@ export class StartTaskComponent implements OnInit {
     @Input()
     appId: number;
 
+    @Input()
+    maxTaskNameLength: number = 255;
+
+    @Input()
+    defaultTaskName: string = '';
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: change to name

File: lib/process-services/task-list/components/start-task.component.ts [hunk 2]
```diff
@@ -88,15 +100,47 @@ export class StartTaskComponent implements OnInit {
...
 
-    public start(): void {
-        if (this.startTaskmodel.name) {
-            if (this.appId) {
-                this.startTaskmodel.category = this.appId.toString();
-            }
-            this.taskService.createNewTask(new TaskDetailsModel(this.startTaskmodel))
+    buildForm() {
+        this.taskModelForm = this.formBuilder.group({
+            taskModelName: new FormControl(this.defaultTaskNameTranslated, [Validators.required, Validators.maxLength(this.maxTaskNameLength)]),
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: we don't need to translate it

File: docs/process-services/start-task.component.md [hunk 3]
```diff
@@ -24,6 +24,11 @@ Creates/Starts new task for the specified app
 | Name | Type | Default value | Description |
 | ---- | ---- | ------------- | ----------- |
 | appId | `number` |  | (required) The id of the app. |
+| taskModelForm | `FormGroup` |  | FormGroup for the new task |
+| taskDetailsModel | `TaskDetailsModel` |  | Model for the task |
+| forms$ | `Observable<Form[]>;` |  | Forms for the dropdown list |
+| maxTaskNameLength | `number` | 100 | Maximum mask name length |
```
Comment (mauriziovitale) [inline docs/process-services/start-task.component.md]: change the default from 100 to 255

File: lib/process-services/task-list/components/start-task.component.ts [hunk 4]
```diff
@@ -45,6 +45,12 @@ export class StartTaskComponent implements OnInit {
     @Input()
     appId: number;
 
+    @Input()
+    maxTaskNameLength: number = 255;
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: we should check that the input is not more than 255

File: docs/process-services/start-task.component.md [hunk 5]
```diff
@@ -24,6 +24,11 @@ Creates/Starts new task for the specified app
 | Name | Type | Default value | Description |
 | ---- | ---- | ------------- | ----------- |
 | appId | `number` |  | (required) The id of the app. |
+| taskModelForm | `FormGroup` |  | FormGroup for the new task |
```
Comment (mauriziovitale) [inline docs/process-services/start-task.component.md]: remove the properties that are not Input

File: lib/process-services/task-list/components/start-task.component.ts [hunk 6]
```diff
@@ -71,6 +77,10 @@ export class StartTaskComponent implements OnInit {
 
     field: FormFieldModel;
 
+    defaultTaskNameTranslated: string;
+
+    taskModelForm: FormGroup;
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: can we change to form or taskForm

File: lib/process-services/task-list/components/start-task.component.ts [hunk 7]
```diff
@@ -88,15 +100,47 @@ export class StartTaskComponent implements OnInit {
         this.preferences.locale$.subscribe((locale) => {
             this.dateAdapter.setLocale(locale);
         });
+        this.defaultTaskNameTranslated = this.translateService.instant(this.defaultTaskName);
         this.loadFormsTask();
+        this.buildForm();
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: move this code to ngOnInit

File: lib/process-services/task-list/components/start-task.component.ts [hunk 8]
```diff
@@ -88,15 +100,47 @@ export class StartTaskComponent implements OnInit {
...
+                    this.setTaskDetailsModel(taskFormData);
+                }
+       });
+    }
+
+    isFormValid() {
+        return this.taskModelForm && this.taskModelForm.valid;
+    }
+
+    setTaskDetailsModel(taskFormData: any) {
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: do we need this method?

File: lib/process-services/task-list/components/start-task.component.ts [hunk 9]
```diff
@@ -88,15 +100,47 @@ export class StartTaskComponent implements OnInit {
...
+        this.formKey = taskFormData.taskModelFormKey;
+
+        this.taskDetailsModel.formKey = taskFormData.taskModelFormKey;
+    }
+
+    public saveTask(): void {
+        if (this.appId) {
+            this.taskDetailsModel.category = this.appId.toString();
+        }
+        if (!this.taskDetailsModel.name) {
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: not needed, can you refactor this method?

File: lib/process-services/task-list/components/start-task.component.ts [hunk 10]
```diff
@@ -178,4 +215,8 @@ export class StartTaskComponent implements OnInit {
             }
         }
     }
+
+    get taskName(): AbstractControl {
```
Comment (mauriziovitale) [inline lib/process-services/task-list/components/start-task.component.ts]: can we call it nameControl

**Comment by proth1** [comment]
Is there a way I can get a preview of the changes being made to the start-task component?

**Pull request closed without merging**